### PR TITLE
Implemented 'sl' command

### PR DESCRIPTION
**Pull request by zlowram**
Implemented the seek-line command. Issue #3814

### CODE REVIEW

**Comment by jvoisin** [comment]
What about factorizing `atoi(input + 2)`?

**Comment by jvoisin** [comment]
Also, indentation is weird.

File: libr/core/cmd_seek.c [hunk 1]
```diff
@@ -263,6 +281,44 @@ static int cmd_seek(void *data, const char *input) {
 			else r_core_seek (core, r_io_desc_size (core->io, core->file->desc), 1);
 			}
 			break;
+		case 'l':
```
Comment (crowell) [inline libr/core/cmd_seek.c]: use tabs

File: libr/core/cmd_seek.c [hunk 2]
```diff
@@ -263,6 +281,44 @@ static int cmd_seek(void *data, const char *input) {
...
+            {
+            ut64 off = UT64_MAX;
+            int curr;
+			if (core->lines_cache == NULL) {
+				r_cons_printf("ERROR: \"lines.from\" and \"lines.to\" must be set\n");
+				break;
+			}
+            switch (input[1]) {
+            case ' ':
+				if (atoi(input+2) < 1 || atoi(input+2) > core->lines_cache_sz-1) {
```
Comment (crowell) [inline libr/core/cmd_seek.c]: dont use `atoi`, use `r_num_math` and cache that value.

File: libr/core/config.c [hunk 3]
```diff
@@ -12,6 +12,42 @@
...
+static int __init_lines_cache(RCore *core, ut64 start_addr, ut64 end_addr) {
+	int i, line_count;
+	int bsz = core->blocksize;
+	char buf[bsz];
+	ut64 off = start_addr;
+
+	if (core->lines_cache != NULL)
+		free(core->lines_cache);
+
+	core->lines_cache = calloc(bsz, sizeof (ut64));
```
Comment (crowell) [inline libr/core/config.c]: R_NEW0

File: libr/core/config.c [hunk 4]
```diff
@@ -12,6 +12,42 @@
...
+		for (i=0; i<bsz; i++) {
+			if (buf[i] == '\n') {
+				core->lines_cache[line_count] = off+i+1;
+				line_count++;
+				if (line_count % bsz == 0) {
+					ut64 *tmp = realloc (core->lines_cache, (line_count+bsz)*sizeof(ut64));
+					if (tmp != NULL) {
+						core->lines_cache = tmp;
+					} else {
+						free(core->lines_cache);
```
Comment (crowell) [inline libr/core/config.c]: R_FREE

**Comment by radare** [comment]
Always space before '('

**Comment by radare** [comment]
I have fixed most of the issues and merged your code. 

Thanks! please review the comments and the current version of the code.

**Pull request closed without merging**